dbms/src/Core/Field.h

```diff
@@ -34,9 +34,23 @@ template <typename T>
 using NearestFieldType = typename NearestFieldTypeImpl<T>::Type;
 
 class Field;
-using Array = std::vector<Field>;
-using TupleBackend = std::vector<Field>;
```

Comment:
Why we needed this before? It seems very confusing and bug-prone.

Reply:
TupleBackend definition looks redundant, but otherwise it's the same thing that I'm doing: have strong typedefs for Array and Tuple fields so that the correct type is deduced by the Field constructor. Actually I'm not even changing this, just renaming: the only place where I changed behavior is in convertFieldToTypeImpl: it won't return Array fields for a Tuple type anymore. Had to re-read this PR to realize what is going on :)

Not sure, maybe it's better to throw away these types and just specify the Field type explicitly.